sdk/python/feast/infra/registry/snowflake.py

```diff
@@ -0,0 +1,1083 @@
+from feast.request_feature_view import RequestFeatureView
+from feast.saved_dataset import SavedDataset, ValidationReference
+from feast.stream_feature_view import StreamFeatureView
+
+
+class FeastMetadataKeys(Enum):
+    LAST_UPDATED_TIMESTAMP = "last_updated_timestamp"
+    PROJECT_UUID = "project_uuid"
+
+
+class SnowflakeRegistryConfig(RegistryConfig):
+    """Registry config for Snowflake"""
+
+    registry_type: Literal["snowflake.registry"] = Field(
+        "snowflake.registry", alias="type"
+    )
+    """ Registry type selector """
+
+    type: Literal["snowflake.registry"] = "snowflake.registry"
+    """ Registry type selector """
+
+    config_path: Optional[str] = os.path.expanduser("~/.snowsql/config")
+    """ Snowflake config path -- absolute path required (Cant use ~) """
+
+    account: Optional[str] = None
+    """ Snowflake deployment identifier -- drop .snowflakecomputing.com """
+
+    user: Optional[str] = None
+    """ Snowflake user name """
+
+    password: Optional[str] = None
+    """ Snowflake password """
+
+    role: Optional[str] = None
+    """ Snowflake role name """
+
+    warehouse: Optional[str] = None
+    """ Snowflake warehouse name """
+
+    authenticator: Optional[str] = None
+    """ Snowflake authenticator name """
+
+    database: StrictStr
+    """ Snowflake database name """
+
+    schema_: Optional[str] = Field("PUBLIC", alias="schema")
+    """ Snowflake schema name """
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class SnowflakeRegistry(BaseRegistry):
+    def __init__(
+        self,
+        registry_config,
+        repo_path,
+    ):
+        assert registry_config is not None and isinstance(
+            registry_config, SnowflakeRegistryConfig
+        ), "SnowflakeRegistry needs a valid registry_config, a path does not work"
+
+        self.registry_config = registry_config
+        self.registry_path = (
+            f'"{self.registry_config.database}"."{self.registry_config.schema_}"'
+        )
+        self.snowflake_conn = get_snowflake_conn(self.registry_config)
+
+        with self.snowflake_conn.cursor() as cur:
+            sql_function_file = f"{os.path.dirname(feast.__file__)}/infra/utils/snowflake/registry/snowflake_table_creation.sql"
+            with open(sql_function_file, "r") as file:
+                sqlFile = file.read()
+
+                sqlCommands = sqlFile.split(";")
+                for command in sqlCommands:
+                    query = command.replace("REGISTRY_PATH", f"{self.registry_path}")
+                    execute_snowflake_statement(cur, query)
+
+        self.cached_registry_proto = self.proto()
+        self.cached_registry_proto_created = datetime.utcnow()
+        self._refresh_lock = Lock()
+        self.cached_registry_proto_ttl = timedelta(
+            seconds=registry_config.cache_ttl_seconds
+            if registry_config.cache_ttl_seconds is not None
+            else 0
+        )
+
+    def refresh(self):
```

Comment:
This is where my issue was for some reason if i gave it an arbitrary second argument feast apply succeeded

Reply:
@sfc-gh-rdervay Im not quite understanding here -- is this about the TTL?